Author: Pino Toscano <pino@debian.org>
Description: Do not bail if $HOME is not set.
 In case there is no $HOME environment variable set (which can happen on
 buildds, for example), do not make cmake error out due to the regexp
 "^$ENV{HOME}.*$".
Forwarded: no
Last-Update: 2011-04-28

--- a/kopete/protocols/skype/skypebuttons/CMakeLists.txt
+++ b/kopete/protocols/skype/skypebuttons/CMakeLists.txt
@@ -2,7 +2,12 @@
 
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=default" )
 
-string( REGEX MATCHALL ^$ENV{HOME}.*$ IS_HOME ${CMAKE_INSTALL_PREFIX} )
+string( LENGTH "$ENV{HOME}" LENGTH_OF_HOME )
+if( LENGTH_OF_HOME GREATER 0 )
+	string( REGEX MATCHALL ^$ENV{HOME}.*$ IS_HOME ${CMAKE_INSTALL_PREFIX} )
+else( LENGTH_OF_HOME GREATER 0 )
+	set( IS_HOME FALSE )
+endif( LENGTH_OF_HOME GREATER 0 )
 string( COMPARE EQUAL /usr ${CMAKE_INSTALL_PREFIX} IS_USR )
 string( COMPARE EQUAL /usr/local ${CMAKE_INSTALL_PREFIX} IS_LOCAL )
 
